stairway/src/test/java/bio/terra/stairway/FlightMapTest.java

```diff
@@ -153,4 +172,28 @@ public void serDesExceptions() {
     Assertions.assertThrows(
         JsonConversionException.class, () -> FlightMap.create(new ArrayList<>(), "garbage"));
   }
+
+  @Test
+  public void typeReferences() {
+    FlightMap flightMap = new FlightMap();
+
+    Map<String, List<UUID>> inputMap = new HashMap<>();
+
+    inputMap.put(
+        "mapKey",
+        new ArrayList<>(
+            Arrays.asList(
+                UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID())));
+
+    final String key = "myList";
+    flightMap.put(key, inputMap);
+    System.out.println(flightMap.getRaw(key));
+
+    Map<String, List<UUID>> outputMap = flightMap.get(key, Map.class);
+    Assertions.assertEquals(inputMap, outputMap);
+
+    HashMap<String, ArrayList<UUID>> explicitOutputMap =
+        flightMap.get(key, new TypeReference<>() {});
+    Assertions.assertEquals(inputMap, explicitOutputMap);
+  }
```

Comment:
On a tangential thread - here at the bottom - the use of `DefaultTyping.NON_FINAL` is deprecated and is viewed as a security risk. It has something to do with a JSON insertion attack that would cause Jackson to execute arbitrary code in a constructor. I don't think we are very vulnerable to that attack: it would require access to the Stairway database to change the JSON. We are not accepting such forms over the wire from random clients.

If you control all of the classes that are serialized, you can fix this with a feature that restricts what can be constructed by a set of classes. For example, if everything was either a basic Java object or derived from StairwayObject. That seemed impractical at the layer where Stairway lives.

Reply:
Is it deprecated? The API we use is, but an equivalent form is not:
```
    ObjectMapper objectMapper = new ObjectMapper();
    objectMapper.activateDefaultTyping(objectMapper.getPolymorphicTypeValidator(),
            ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.WRAPPER_ARRAY);
```